### PR TITLE
CI: Travis -> Github Actions

### PR DESCRIPTION
**Pull request by junghans**
Initial step in migrating from travis to GHA. WIll continue with #345

### CODE REVIEW

**Comment by junghans** [comment]
@streeve some cajita tests fail on `fedora:rawhide` any ideas?

**Comment by streeve** [comment]
I'm trying to recreate locally, but no luck yet

**Comment by sslattery** [comment]
> @streeve some cajita tests fail on `fedora:rawhide` any ideas?

Man `IndexConversion` is a weird test to fail. Perhaps there is an uninitialized value issue somewhere I missed because it is dumping out `0`?

**Comment by junghans** [comment]
> I'm trying to recreate locally, but no luck yet

Did you try the docker container?
```
docker run -it ghcr.io/kokkos/ci-containers/fedora:rawhide /bin/bash
```

**Comment by streeve** [comment]
> Did you try the docker container?

Turns out that's super helpful. 

`fedora:rawhide` tests seem fine - only failing because of mpi (passed with `-oversubscribe`)
I reproduced the error with `fedora:latest`; it only affects periodic cases, but no fix yet

**Comment by streeve** [comment]
So `Cajita_Halo` failure is the issue (causing the periodic `Cajita_IndexConversion` failures

**Comment by streeve** [comment]
I haven't found the bug that causes `Halo` to fail on some distributions, but this now at least runs all the tests. I had to reduce printing in those failing tests (not sure why GHA doesn't truncate) and remove `ubuntu:devel` for now as it hangs

File: .github/workflows/CI2.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }}
+          cmake --build build --parallel 2
+          cmake --install build
+      - name: Checkout Cabana
+        uses: actions/checkout@v2.2.0
+      - name: Build Cabana
+        run: |
+          cmake -B build \
+            -DCMAKE_INSTALL_PREFIX=$HOME/Cabana \
+            -DMPIEXEC_MAX_NUMPROCS=2 -DMPIEXEC_PREFLAGS="--oversubscribe" \
```
Comment (sslattery) [inline .github/workflows/CI2.yml:76]: Is 2 MPI ranks our limit here? Do we just get one physical core per build? Would be nice to do 4 if possible - even if we oversubscribe in terms of MPI.

File: cajita/unit_test/tstHalo.hpp [hunk 2]
```diff
@@ -54,6 +54,7 @@ int haloPad( Edge<D>, int d )
 template <class Array>
 void checkGather( const int halo_width, const Array& array )
 {
+    int pass_gather_test = 1;
```
Comment (sslattery) [inline cajita/unit_test/tstHalo.hpp:57]: Were these changes part of trying to address the failing tests?

**Review by sslattery — APPROVED** [review]
Just a few basic questions. I see the Travis still showing up in the branch protection even though the `.travis.yml` has been removed - is there something else in the repo settings we need to change?

File: .github/workflows/CI2.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    continue-on-error: ${{ matrix.kokkos_ver == 'develop' || matrix.distro == 'fedora:latest' || matrix.distro == 'fedora:intel' || matrix.distro == 'ubuntu:devel' }}
+    strategy:
+      matrix:
+        distro: ['fedora:latest', 'fedora:rawhide', 'opensuse:latest', 'ubuntu:latest', 'ubuntu:rolling'] # 'ubuntu:devel'
+        cxx: ['g++', 'clang++']
+        openmp: ['ON']
+        cmake_build_type: ['Debug', 'Release']
+        kokkos_ver: ['3.2.01', 'master', 'develop']
+        include:
+          - distro: 'fedora:intel'
```
[the remark this answers is not in the document]
Comment (streeve) [reply .github/workflows/CI2.yml:23]: line 17 adds `distro` for the full matrix with all options; those in `include` are more of a spot check - we could change which distributions are in which

File: cajita/unit_test/tstHalo.hpp (hunk 2 above)
Comment (streeve) [reply cajita/unit_test/tstHalo.hpp:57]: Yes, just limiting the prints because it was hanging

**Comment by streeve** [comment]
> I see the Travis still showing up in the branch protection even though the `.travis.yml` has been removed - is there something else in the repo settings we need to change?

I don't have those permissions, but yes I think there's a setting to require Travis

File: cajita/unit_test/tstHalo.hpp (hunk 2 above)
Comment (streeve) [reply cajita/unit_test/tstHalo.hpp:57]: Do you want a `fixme` for each?

File: .github/workflows/CI2.yml (hunk 1 above)
Comment (streeve) [reply .github/workflows/CI2.yml:76]: It must be one if we couldn't run without this. Maybe change that separately?

**Comment by sslattery** [comment]
> > I see the Travis still showing up in the branch protection even though the `.travis.yml` has been removed - is there something else in the repo settings we need to change?
> 
> I don't have those permissions, but yes I think there's a setting to require Travis

OK - I'll add you to the list. We don't wan't to disable anyway until this one is merged. All other open PRs will also need to be rebased to get the new CI

File: cajita/unit_test/tstHalo.hpp (hunk 2 above)
Comment (sslattery) [reply cajita/unit_test/tstHalo.hpp:57]: yes let's add this. Seeing as it was distribution dependent we will really want to dig.

File: .github/workflows/CI2.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (junghans) [reply .github/workflows/CI2.yml:23]: Yeah, I didn't wanted to build the full matrix.

File: .github/workflows/CI2.yml (hunk 1 above)
Comment (junghans) [reply .github/workflows/CI2.yml:76]: Yeah, I think github actions has 4 cores, but let's bump that in an extra PR.

File: .github/workflows/CI2.yml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            cxx: 'icpc'
+            openmp: 'ON'
+            cmake_build_type: 'Debug'
+            kokkos_ver: 'master'
+          - distro: 'fedora:intel'
+            cxx: 'icpc'
+            cmake_build_type: 'Release'
+            openmp: 'OFF'
+            kokkos_ver: 'master'
+            cxx_flags: ' -parallel'
```
Comment (junghans) [inline .github/workflows/CI2.yml:38]: Why is that needed? Especially because openmp is `OFF`.

Comment (streeve) [reply .github/workflows/CI2.yml:38]: It's only for the `peakflops` tests. So something about vectorization

Comment (junghans) [reply .github/workflows/CI2.yml:38]: I see, might be something to document.
